engines/python/setup/djl_python/huggingface.py

```diff
@@ -496,41 +496,120 @@ def register_adapter(inputs: Input):
-                                                       adapter_path,
-                                                       adapter_name)
+        raise NotImplementedError(
+            "LoRA adapter API is only supported for rolling batch.")
+
+    adapter_name = inputs.get_property("name")
+    adapter_path = inputs.get_property("src")
+    adapter_pin = inputs.get_property(
+        "pin").lower() == "true" if inputs.get_property("pin") else False
+
+    if adapter_name not in _service.adapter_registry:
+        raise ValueError(f"Adapter {adapter_name} not registered.")
+
+    try:
+        adapter_old = _service.adapter_registry[adapter_name]
+        if adapter_old.get_property("src") and adapter_old.get_property(
+                "src") != adapter_path:
+            raise NotImplementedError(
+                f"Updating adapter path is not supported.")
 
-        if isinstance(_service.hf_pipeline, Pipeline):
-            _service.hf_pipeline.model = _service.model
+        if adapter_old.get_property("pin") and adapter_old.get_property(
+                "pin").lower() == "true" and not adapter_pin:
+            raise NotImplementedError(f"Unpin adapter is not supported.")
 
-        if isinstance(_service.hf_pipeline_unwrapped, Pipeline):
-            _service.hf_pipeline_unwrapped.model = _service.model
-    return Output()
+        _service.adapter_registry[adapter_name] = inputs
+
+        if adapter_pin:
+            pinned = _service.rolling_batch.pin_lora(adapter_name)
+            if not pinned:
+                raise RuntimeError(
+                    f"Failed to pin LoRA adapter {adapter_name}")
+    except Exception as e:
+        if any(msg in str(e)
+               for msg in ("No free lora slots",
+                           "greater than the number of GPU LoRA slots")):
+            raise MemoryError(str(e))
+        return Output().error("update_adapter_error", message=str(e))
+
+    logging.info(f"Updated adapter {adapter_name} successfully")
+    return Output(message=f"Adapter {adapter_name} updated")
 
 
 def unregister_adapter(inputs: Input):
     """
     Unregisters lora adapter from the model.
     """
+    if not is_rolling_batch_enabled(_service.hf_configs.rolling_batch):
+        raise NotImplementedError(
+            "LoRA adapter API is only supported for rolling batch.")
+
     adapter_name = inputs.get_property("name")
-    logging.info(f"Unregistering adapter {adapter_name}")
-    #TODO: delete in vllm engine as well
+
+    if adapter_name not in _service.adapter_registry:
+        raise ValueError(f"Adapter {adapter_name} not registered.")
     del _service.adapter_registry[adapter_name]
-    if not is_rolling_batch_enabled(_service.hf_configs.rolling_batch):
-        _service.model.base_model.delete_adapter(adapter_name)
-    return Output()
+
+    try:
+        removed = _service.rolling_batch.remove_lora(adapter_name)
+        if not removed:
+            logging.debug(
```

Comment:
let's make this an info level for now - most endpoints are going to be deploying at that level and would be good to know initially how frequently this occurs

Reply:
Done.